src/hex/tests.rs

```diff
@@ -337,6 +337,13 @@ fn line_to() {
     );
 }
 
+#[test]
+fn empty_line_to() {
+    let start = Hex::new(3, -7);
+    let line: Vec<_> = start.line_to(start).collect();
+    assert_eq!(line, vec![start, start]);
```

Comment:
This seems like the wrong result still: I would expect a vec of length one, with only `start` as the element.